domains/environment_domains/front_door.tf

```diff
@@ -22,12 +22,90 @@ resource "azurerm_cdn_frontdoor_origin_group" "main" {
 }
 
 resource "azurerm_cdn_frontdoor_origin" "main" {
-  name                          = "${var.environment}-org"
-  cdn_frontdoor_origin_group_id = azurerm_cdn_frontdoor_origin_group.main.id
+  name                           = "${var.environment}-org"
+  cdn_frontdoor_origin_group_id  = azurerm_cdn_frontdoor_origin_group.main.id
   certificate_name_check_enabled = true
-  host_name = var.host_name
-  origin_host_header = var.host_name
+  enabled                        = true
+  host_name                      = var.host_name
+  origin_host_header             = var.host_name
 }
 
-# TODO: Create custom domain
-# TODO: Create route
+resource "azurerm_cdn_frontdoor_custom_domain" "main" {
+  for_each                 = toset(var.domains)
+  name                     = each.key
+  cdn_frontdoor_profile_id = data.azurerm_cdn_frontdoor_profile.main.id
+  dns_zone_id              = data.azurerm_dns_zone.main.id
+  host_name                = each.key == "apex" ? "${var.zone}" : "${each.key}.${var.zone}"
+  tls {
+    certificate_type    = "ManagedCertificate"
+    minimum_tls_version = "TLS12"
+  }
+}
+
+resource "azurerm_dns_txt_record" "main" {
+  for_each            = { for k, v in toset(var.domains) : k => v if v != "apex" }
+  name                = join(".", ["_dnsauth", "${each.key}"])
```

Comment:
Won't the `key` be the index of each item in the set, ie an int when we want the string value that corresponds to the subdomain as described [here](https://learn.microsoft.com/en-us/azure/frontdoor/front-door-how-to-onboard-apex-domain?pivots=front-door-standard-premium#onboard-the-custom-domain-to-your-front-door)?

Reply:
key appears to be the string value. i.e apex and www in ["apex", "www"].